app/components/ContextMenu/index.tsx

```diff
@@ -50,20 +52,35 @@ export default function ContextMenu({
   const previousVisible = usePrevious(rest.visible);
   const maxHeight = useMenuHeight(rest.visible, rest.unstable_disclosureRef);
   const backgroundRef = React.useRef<HTMLDivElement>(null);
+  const { ui } = useStores();
+  const setIsMenuOpen = useSidebar();
 
   React.useEffect(() => {
     if (rest.visible && !previousVisible) {
       if (onOpen) {
         onOpen();
       }
+      if (ui.sidebarCollapsed && setIsMenuOpen) {
```

Comment:
I'm not sure why `ui.sidebarCollapsed` is considered here, the menu is open either way.